assets/sass/02-tools/mixins.scss

```diff
@@ -77,3 +77,21 @@
 		color: var(--button--color-text-active);
 	}
 }
+
+@mixin innerblock-margin-clear() {
+
+	// Clear the top margin for the first-child.
+	> *:first-child {
+		margin-top: 0;
+	}
+
+	// Last child that is not the appender.
+	> *:last-child():not(.block-list-appender) {
+		margin-bottom: 0;
+	}
+
+	// When selected, the last item becomes the second last because of the appender.
+	&.has-child-selected > *:nth-last-child(2) {
```

Comment:
This isn't applied correctly to Groups, Media & Text, or Cover blocks, because the `has-child-selected` class is added to a parent container (ex, `.wp-block-media-text`, not `.wp-block-media-text__content`). It's the right element for a single column though, so changing this to `.has-child-selected &` wouldn't work across all cases (it also doesn't work for Media & Text as-is, since it becomes `.has-child-selected .wp-block-media-text__content .wp-block-media-text__content ...`)

This causes a pretty obvious content jump if you've got content bottom-aligned, so it would be nice to figure out something here 🤔 

![margin-jump](https://user-images.githubusercontent.com/541093/99838380-10cd0780-2b37-11eb-9d47-d2afe4cb1238.gif)

Reply:
In 37be8a3 I reworked the mixin to account for all these various containers. It _mostly_ works, but there are two bugs: 

- Sometimes the styles don't kick in immediately on click — I think this is because of a delay between when you select a block, and when `.has-child-selected` is added. 
- Sometimes the spacing still jumps around a little bit because of the appender taking up space. This is what's happening in your GIF above).

Both of these existed before this PR, and I'm not totally sure how we'd solve them without styling the appender itself, which seems unwise.

In general, it doesn't look like Gutenberg zeroes any of these margins on its own... another route would be remove all this entirely and just let Gutenberg handle it eventually (if that ever happens). I think we're getting closer and closer to "weird css hack" territory here. 😅

Reply:
Yeah, I'm seeing both bugs you mentioned. The content jump-up is smaller at least, rather than the full 38px of margin + appender, it's just the appender (16px) now.

> another route would be remove all this entirely and just let Gutenberg handle it eventually (if that ever happens). I think we're getting closer and closer to "weird css hack" territory here.

Yeahhh, we definitely are. If we remove the mixin, we end up with extra space between blocks - but it's the same amount of extra space, not the uneven alignment we were seeing before. It does totally get rid of the content jump.